pandas/core/indexes/period.py

```diff
@@ -783,6 +788,9 @@ def get_indexer(self, target, method=None, limit=None, tolerance=None):
 
         if tolerance is not None:
             tolerance = self._convert_tolerance(tolerance)
+            if target.size != tolerance.size and tolerance.size > 1:
```

Comment:
i think would also pass target to _convert_tolerance so
can move error checks there

Reply:
you would change all _convert_tolerance to pass target as well
to